### PR TITLE
lower replication_pad3d and replication_pad3d_backward

### PR DESCRIPTION
**Pull request by ManfeiBai**
follow up Reenable PR for https://github.com/pytorch/xla/pull/6537 and #6554

---
passed local test:
```
# pytest test/test_core_aten_ops.py -k test_aten_replication_pad3d_0
=============================== test session starts ===============================
platform linux -- Python 3.10.13, pytest-8.0.0, pluggy-1.4.0
rootdir: /root/pytorch
configfile: pytest.ini
plugins: hypothesis-6.97.4
collected 499 items / 498 deselected / 1 selected                                 

test/test_core_aten_ops.py WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1708454923.760699  843883 pjrt_api.cc:100] GetPjrtApi was found for tpu at /root/miniconda3/envs/torch310/lib/python3.10/site-packages/libtpu/libtpu.so
I0000 00:00:1708454923.760777  843883 pjrt_api.cc:79] PJRT_Api is set for device type tpu
I0000 00:00:1708454923.760792  843883 pjrt_api.cc:146] The PJRT plugin has PJRT API version 0.40. The framework PJRT API version is 0.40.
.                                                [100%]

======================== 1 passed, 498 deselected in 6.46s ========================
# pytest test/test_core_aten_ops.py -k test_aten_replication_pad3d_1
=============================== test session starts ===============================
platform linux -- Python 3.10.13, pytest-8.0.0, pluggy-1.4.0
rootdir: /root/pytorch
configfile: pytest.ini
plugins: hypothesis-6.97.4
collected 499 items / 498 deselected / 1 selected                                 

test/test_core_aten_ops.py WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1708454989.804666  845455 pjrt_api.cc:100] GetPjrtApi was found for tpu at /root/miniconda3/envs/torch310/lib/python3.10/site-packages/libtpu/libtpu.so
I0000 00:00:1708454989.804751  845455 pjrt_api.cc:79] PJRT_Api is set for device type tpu
I0000 00:00:1708454989.804773  845455 pjrt_api.cc:146] The PJRT plugin has PJRT API version 0.40. The framework PJRT API version is 0.40.
.                                                [100%]

======================== 1 passed, 498 deselected in 6.60s ========================
```

---
print metric locally: https://gist.github.com/ManfeiBai/e661eab6fae8a10a1828369a2a016b8e

### CODE REVIEW

**Comment by ManfeiBai** [comment]
CI failed due to `ERROR: Failed to query remote execution capabilities: Unexpected error refreshing access token`

**Review by wonjoolee95 — COMMENTED** [review]
Thanks, @ManfeiBai! I didn't know we can directly re-use the existing replication pad lowering logic, but it seems like we can. Nice!

Also why did we lower the `_backward` variant? Is it required?

**Comment by ManfeiBai** [comment]
> _backward

yes, we don't have test in `test/test_core_aten_ops.py` for backward, we lower the `_backward` variant due to `replication_pad1d` and `replication_pad2d` also did that, do I add the same for `replication_pad3d` and the matched CPP tests here too

**Review by wonjoolee95 — APPROVED** [review]
SGTM, thanks! Feel free to include the newline change with another PR, don't want to block this PR for a nit comment.

File: torch_xla/csrc/tensor_methods.h [hunk 1]
```diff
@@ -738,6 +738,12 @@ XLATensorPtr replication_pad2d_backward(const XLATensorPtr& grad_output,
                                         const XLATensorPtr& input,
                                         std::vector<int64_t> padding);
 
+XLATensorPtr replication_pad3d(const XLATensorPtr& input,
+                               std::vector<int64_t> padding);
+XLATensorPtr replication_pad3d_backward(const XLATensorPtr& grad_output,
```
[the remark this answers is not in the document]
Comment (ManfeiBai) [reply torch_xla/csrc/tensor_methods.h:743]: Thanks, Wonjoo, according to https://github.com/pytorch/xla/blob/0192ff75324d51d748d76f7717bbccabc15d1db8/torch_xla/csrc/tensor_methods.h#L729C1-L745C71, do we want to keep the same style like 1d and 2d without empty newline here?

[the remark this answers is not in the document]
Comment (wonjoolee95) [reply torch_xla/csrc/tensor_methods.h:743]: Ah, then let's just keep it as is. Thanks!

**Pull request merged**